### PR TITLE
Add initial type="" API draft

### PR DESCRIPTION
**Pull request by domenic**
This is one take on the discussions in #18, that makes the existing text more concrete. The issue will remain open to discuss these details as well as alternatives.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -176,9 +176,21 @@ though this timeout will be suspended while the toast has focus or the mouse is
...
-- `type`: category of toast, conveying the semantic priority of the toast, and influencing its styling (both default and user-provided). This will correspond closely with certain [WAI-ARIA roles](https://w3c.github.io/using-aria/#aria-roles) 
-(more discussion in [#18](https://github.com/jackbsteinberg/std-toast/issues/18)).
-    - TODO([#18](https://github.com/jackbsteinberg/std-toast/issues/18)): decide on list of WAI-ARIA Roles to use, and decide on types to natively support, create, and style.
+- `type`: an [enumerated attribute](https://html.spec.whatwg.org/multipage/common-microsyntaxes.html#enumerated-attribute) giving the type of toast:
+  one of `"info"`, `"success"`, `"warning"`, or `"error"`.
+  This is used to convey the [semantics](https://html.spec.whatwg.org/multipage/dom.html#represents) of the toast,
+  similar to the distinctions between e.g. `<ol>` / `<ul>` / `<menu>` or `<strong>` / `<em>` / `<small>`.
+  - Like other semantic distinctions,
+    authors may want to style based on the distinction.
+    The [default styles](#default-styles) only change the border color of the toast,
```
Comment (jackbsteinberg) [inline README.md:186]: I like this for the time being, but we definitely want to remember to bring this up in other toast default style conversations (this is a lot subtler than most other toast libraries' type styles)

File: README.md [hunk 2]
```diff
@@ -176,9 +176,21 @@ though this timeout will be suspended while the toast has focus or the mouse is
...
+    but can be overridden by a page or design system to give any desired appearance.
+  - The default type is `"info"`.
+  - By default, `"error"` toasts will be treated as having the ARIA role semantics of [alert](https://rawgit.com/w3c/aria/master/#alert),
```
Comment (jackbsteinberg) [inline README.md:188]: I feel like `"warning"` toasts should also have the alert role (assuming we're adding aria roles) but that's a personal preference, either could work

File: README.md [hunk 3]
```diff
@@ -209,6 +221,9 @@ const toast = document.createElement('std-toast');
 console.log(toast.open); // false
 ```
 
+The `type` enumerated attribute is reflected [limited to only known values](https://html.spec.whatwg.org/multipage/common-dom-interfaces.html#limited-to-only-known-values),
```
Comment (jackbsteinberg) [inline README.md:224]: In the text this shows up looking like "the `type` enumerated attribute is reflected limited to only known values", there should probably be parens around the `limited to only known values` portion.

File: README.md [hunk 4]
```diff
@@ -176,9 +176,21 @@ though this timeout will be suspended while the toast has focus or the mouse is
 - [Global attributes](https://html.spec.whatwg.org/multipage/dom.html#global-attributes)
 - `open`: a boolean attribute, determining whether the toast is visible or not (according to the default styles).
 By default toasts are not shown.
-- `type`: category of toast, conveying the semantic priority of the toast, and influencing its styling (both default and user-provided). This will correspond closely with certain [WAI-ARIA roles](https://w3c.github.io/using-aria/#aria-roles) 
-(more discussion in [#18](https://github.com/jackbsteinberg/std-toast/issues/18)).
-    - TODO([#18](https://github.com/jackbsteinberg/std-toast/issues/18)): decide on list of WAI-ARIA Roles to use, and decide on types to natively support, create, and style.
+- `type`: an [enumerated attribute](https://html.spec.whatwg.org/multipage/common-microsyntaxes.html#enumerated-attribute) giving the type of toast:
+  one of `"info"`, `"success"`, `"warning"`, or `"error"`.
+  This is used to convey the [semantics](https://html.spec.whatwg.org/multipage/dom.html#represents) of the toast,
```
Comment (jackbsteinberg) [inline README.md]: Thought: should the types be announced differently to screen readers?

For instance with this toast:

```html
<std-toast type="error" open>
  Access Denied!
</std-toast>
```

Should the screen reader read out something like `error: Access Denied!`? Or is this encompassed by the toasts' aria roles?

**Comment by jackbsteinberg** [comment]
One issue that came through when exploring the implementation of this API update is that, without an easy way to fallback to a `type="info"` value, the pattern doesn't make sense. I propose we update the spec language to represent four toast types, `success`, `warning`, `error`, and a default type, the last of which is not explicit and is by definition the absence of type.

The explicit types would could be styled using `[type=____ i]` selectors, and the default type would be styled directly onto the toast. This way, the default is more of a fallback, absence of type itself, instead of its own distinct type, which would be technically confusing to implement.

File: README.md (hunk 2 above)
Comment (domenic) [reply README.md:188]: Let's leave this pending discussion in #18. I'm inclined to start conservatively.

**Comment by domenic** [comment]
That makes sense. In particular, having `<std-toast></std-toast>` behave mostly the same as `<std-toast type="info"></std-toast>`, except that only the latter works with `std-toast[type=info i] { ... }` styling, is kind of bad. I'll update to remove the explicit info value for the type="" attribute.

File: README.md (hunk 4 above)
Comment (domenic) [reply README.md]: My general understanding is that screen-reader users should not get content tailored only for them in this way, but I have to run for tonight and want to do more reading before making a definitive statement in that regard.

**Pull request merged**